node/network/bridge/src/validator_discovery.rs

```diff
@@ -278,22 +303,24 @@ impl<N: Network, AD: AuthorityDiscovery> Service<N, AD> {
 		(network_service, authority_discovery_service)
 	}
 
+	/// Should be called when a peer connected.
 	pub async fn on_peer_connected(&mut self, peer_id: &PeerId, authority_discovery_service: &mut AD) {
 		// check if it's an authority we've been waiting for
 		let maybe_authority = authority_discovery_service.get_authority_id_by_peer_id(peer_id.clone()).await;
 		if let Some(authority) = maybe_authority {
 			for request in self.non_revoked_discovery_requests.iter_mut() {
 				let _ = request.on_authority_connected(&authority, peer_id);
 			}
-			let _ = self.connected_validators.insert(authority, peer_id.clone());
+
+			self.connected_peers.entry(peer_id.clone()).or_default().insert(authority);
+		} else {
+			self.connected_peers.insert(peer_id.clone(), Default::default());
```

Comment:
why do we need to insert it? do we differentiate between a connected `PeerId` without known `AuthorityId`s from a disconnected `PeerId`?

Reply:
Because this tracks all connected peers, not only validators.